File: backend/Contracts/DF.Contracts/Gateway/Responses/Dish/DishResponse.cs
using System;
using System.Collections.Generic;
using DF.Contracts.Enums;

namespace DF.Contracts.Gateway.Responses.Dish;

public record DishResponse(
    Guid Id, 
    string Name, 
    string? Description, 
    string? ImageUrl, 
    decimal Price, 
    Category Category,
    int CookingTime,
    List<IngredientResponse> Ingredients);
    
public record DishForCustomerResponse(
    Guid Id, 
    string Name, 
    string? Description, 
    string? ImageUrl, 
    decimal Price, 
    Category Category,
    int CookingTime,
    BusinessResponse BusinessDetails,
    List<IngredientResponse> Ingredients);

File: frontend/food-delivery-platform/src/models/responses/dish/DishResponse.ts
import {Category} from "../../../constants/category";
import {IngredientResponse} from "./IngredientResponse";

export interface DishResponse {
    id: string
    name: string
    description?: string | null
    imageUrl?: string | null
    price: number
    category: Category
    cookingTime: number
    ingredients: IngredientResponse[]
}
